src/services/UserValidationService.ts

```diff
@@ -1,42 +1,57 @@
 import { Service } from '@tsed/di';
 import { Unauthorized } from '@tsed/exceptions';
-import * as bcrypt from 'bcrypt';
 
-import { User } from '../entities/User';
+import { UserCreation } from '../models/UserCreation';
+import { UserLogin } from '../models/UserLogin';
 
 @Service()
 export class UserValidationService {
 
-	public async validateUser(
-		userToValidate: User, 
-	): Promise<User> {
+	public validateUserLogin(userLogin: UserLogin): void {
+		const email: string= userLogin.email;
+		const password: string = userLogin.password;
+		
+		this.checkParameters([email, password]);
+		this.checkEmail(email);
+		this.checkPassword(password);
+	}
+
+	public validateUserCreation(userCreation: UserCreation): void {
 
-		const username: string = userToValidate.username;
-		const email: string= userToValidate.email;
-		const password: string = userToValidate.password;
+		const username: string = userCreation.username;
+		const email: string= userCreation.email;
+		const password: string = userCreation.password;
 		
-		if (Object.keys(userToValidate).length === 0) {
-			throw new Unauthorized("Empty request body!");
-		}
-	
-		if (!username || !email || !password) {
-			throw new Unauthorized("Missing credentials!");
-		}
+		this.checkParameters([username, email, password]);
+		this.checkUserName(username);
+		this.checkEmail(email);
+		this.checkPassword(password);
+	}
 
+	private checkParameters(parameters: string[]): void {
+		parameters.forEach(s => {
+			if (!s) {
+				throw new Unauthorized("Missing credentials!");
+			}
+		});
+	}
+
+	private checkUserName(username: string): void {
 		if (!this.containWhitespaces(username)) {
 			throw new Unauthorized("Whitespaces in username!");
 		}
+	}
 
+	private checkEmail(email: string): void {
 		if (!this.isValidEmailFormat(email)) {
 			throw new Unauthorized("Invalid email format!");
 		}
+	}
 
+	private checkPassword(password: string): void {
 		if (!this.isStrongPassword(password)) {
 			throw new Unauthorized("Week password!");
 		}
-
-		userToValidate.password = await this.encryptPassword(password);
-		return userToValidate;
 	}
 
 	private containWhitespaces(string: string): boolean {
```

Comment:
I've probably missed this, but a better naming would be `hasWhitespace`.

The reason for that is the following:
boolean is a true or false value. In other words, you always get a `yes` or `no` response. Because of that, it is often considered good practice to name your variables/functions by what they do. If a property is boolean or it is a function returning boolean, whenever you retrieve it's value, you will get `yes` or `no` answer.

In that spirit, if you ask `does this string have whitespaces?`, you will get a `yes` or `no` answer. Of course, we shouldn't really use that long question, but we can simplify it with `hasWhitespace`. Whenever you are returning different values, the naming might be quite hard to figure. Often times, using `get` prefix is the best you can do.
 
Similarly, when you are executing a `void` function, the naming should be in imperative way.

If you are reacting to some events, for example user click, it is also good practice to use `on` prefix for every event handler. `onButtonClick()`, `onFormSubmit()` or any other event is more easily found.

```
function isPasswordValid(password: string): boolean { ... } // returns a boolean
function validatePassword(password: string): void { ... } // Don't return value, do something inside the function
function getEncryptedPassword(password: string): string { ... } // returns an encrypted password
```

Similarly to other comments, I'll approve this, but I believe it is good to have this written down.

Reply:
Ok this I will also change. Really thanks for that btw. I still have big problem with good, clean and clear naming. I need this a lot.